### PR TITLE
Add important Timeline events to whitelist

### PR DESCRIPTION
**Pull request by B3rn475**
By marking these events they will be shown in the new developer centric Timeline Dashboard page in Observatory.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/object.dart [hunk 1]
```diff
@@ -126,7 +126,8 @@ class PaintingContext {
   void paintChild(RenderObject child, Offset offset) {
     assert(() {
       if (debugProfilePaintsEnabled)
-        Timeline.startSync('${child.runtimeType}');
+        Timeline.startSync('${child.runtimeType}',
+          arguments: const <String, String>{ 'mode': 'basic' });
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: nit: if blocks with multiple lines need braces.

Also, please use Flutter-style line wrapping (newline after the `(`, newline before the `)`) if you wrap to multiple lines.

In this case what you might want to do is declare this constant in foundation/debug.dart, and just use that here. That'll avoid a lot of duplicate code and also will let you not wrap things to two or more lines.

**Comment by Hixie** [comment]
Does Observatory have end-to-end integration tests? And if so, is there any way we can use those to test that this works as expected?

File: packages/flutter/lib/src/rendering/object.dart (hunk 1 above)
Comment (B3rn475) [reply packages/flutter/lib/src/rendering/object.dart]: done

**Comment by B3rn475** [comment]
While Observatory is tested there are no end-to-end tests for this particular case.

**Comment by Hixie** [comment]
What I really meant was is there enough infrastructure in place that it would make sense to add more so that we can test this. This is the kind of thing that could very easily regress since it depends on a lot of finicky details to get right (all the port forwarding has to be right, the protocol has to be right, the various Dart versions have to be close to in sync, the framework has to send the events the right way, etc).

**Comment by Hixie** [comment]
LGTM with extra bonus points if you could find a way to actually test this end-to-end.

**Comment by B3rn475** [comment]
I'll open an issue on `dart-lang/sdk`, we need to find a reliable way to test the timeline for this use-case.

**Comment by Hixie** [comment]
Was it expected that this would reduce performance by 15% on some benchmarks?

**Comment by B3rn475** [comment]
No, it was not

**Comment by rmacnak-google** [comment]
The use of arguments causes the timeline to miss a few fast paths in the timeline implementation. Still I wouldn't expect such a large effect.

**Pull request merged**